src/components/authentication/Signup.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable no-alert */
+/* eslint-disable no-undef */
+/* eslint-disable no-console */
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { toast } from 'react-toastify';
+import {
+  Container, Form, Col, Button, Image, Row,
+} from 'react-bootstrap';
+import { signupAction } from '../../redux/actions/signupAction';
+import '../../scss/signup.scss';
+import BackgroundImage from '../../assets/backg.jpg';
+import FormItem from './FormItem';
+
+
+const Signup = (props) => {
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
```

Comment:
This file is too big! I'm finding it hard to follow thru the flow, please consider moving certain parts like useState hook variable to external utils and refactoring if-statements.